pkg/ctl/cmdutils/scale_test.go

```diff
@@ -33,171 +33,208 @@ var _ = Describe("scale node group config file loader", func() {
+				name:        "ng-with-max",
+				desiredSize: aws.Int(2),
+				minSize:     aws.Int(3),
+				err:         fmt.Errorf("minimum number of nodes must be fewer than or equal to number of nodes"),
+			}),
+			Entry("min greater than max", scaleNodeGroupCLICase{
+				name:    "ng-with-max",
+				minSize: aws.Int(3),
+				maxSize: aws.Int(2),
+				err:     fmt.Errorf("maximum number of nodes must be greater than minimum number of nodes"),
+			}),
+			Entry("not specifying any", scaleNodeGroupCLICase{
+				name: "ng-with-max",
+				err:  fmt.Errorf("at least one of minimum, maximum and desired nodes must be set"),
+			}),
+		)
+
+		Describe("for managed nodegroups", func() {
+			Context("when using a config file", func() {
+				It("setting --name finds that individual nodegroup", func() {
+					ngName := "mng-ng"
+					ng := &api.NodeGroupBase{
+						Name: ngName,
+						ScalingConfig: &api.ScalingConfig{
+							DesiredCapacity: aws.Int(2),
+						},
+					}
+
+					config := api.NewClusterConfig()
+					config.Metadata.Name = "test-cluster"
+					config.ManagedNodeGroups = []*api.ManagedNodeGroup{
+						{
+							NodeGroupBase: &api.NodeGroupBase{
+								Name: ngName,
+								ScalingConfig: &api.ScalingConfig{
+									DesiredCapacity: aws.Int(2),
+								},
+							},
+						},
+					}
+					cmd := &Cmd{
+						CobraCommand:   newCmd(),
+						ClusterConfig:  config,
+						ProviderConfig: api.ProviderConfig{},
+					}
+
+					err := NewScaleNodeGroupLoader(cmd, ng).Load()
+					Expect(err).NotTo(HaveOccurred())
+				})
+			})
+		})
+	})
+
+	When("scaling all nodegroups", func() {
+		When("using a config file", func() {
+			It("finds all nodegroups", func() {
+				cmd := &Cmd{
+					CobraCommand:      newCmd(),
+					ClusterConfigFile: "test_data/scale-all-ng-test.yaml",
+				}
+
+				err := NewScaleAllNodeGroupLoader(cmd).Load()
 				Expect(err).NotTo(HaveOccurred())
+				Expect(len(cmd.ClusterConfig.ManagedNodeGroups)).To(Equal(1))
+				Expect(*cmd.ClusterConfig.ManagedNodeGroups[0].BaseNodeGroup().ScalingConfig.DesiredCapacity).To(Equal(3))
+				Expect(*cmd.ClusterConfig.ManagedNodeGroups[0].BaseNodeGroup().ScalingConfig.MinSize).To(Equal(3))
+				Expect(*cmd.ClusterConfig.ManagedNodeGroups[0].BaseNodeGroup().ScalingConfig.MaxSize).To(Equal(5))
+
+				Expect(len(cmd.ClusterConfig.NodeGroups)).To(Equal(1))
+				Expect(*cmd.ClusterConfig.NodeGroups[0].BaseNodeGroup().ScalingConfig.DesiredCapacity).To(Equal(3))
+				Expect(*cmd.ClusterConfig.NodeGroups[0].BaseNodeGroup().ScalingConfig.MinSize).To(Equal(3))
+				Expect(*cmd.ClusterConfig.NodeGroups[0].BaseNodeGroup().ScalingConfig.MaxSize).To(Equal(5))
+			})
+		})
+
+		When("not using a config file", func() {
+			It("fails", func() {
+				cmd := &Cmd{
+					CobraCommand: newCmd(),
+				}
+
+				err := NewScaleAllNodeGroupLoader(cmd).Load()
+				Expect(err).To(HaveOccurred())
+				Expect(err).To(MatchError("a config file is required when scaling multiple nodegroups"))
 			})
 		})
```

Comment:
what is this scenario :thinking: ? `eksctl scale nodegroup --name foo`? The error message implies your using a config file, but your not passing one in so perhaps a more generic error is better?

Reply:
I updated this file, hopefully the scenarios are a bit clearer now! Also added a couple of missing unit tests :)